server/sonar-db-migration/src/test/java/org/sonar/server/platform/db/migration/version/v67/CleanupDisabledUsersTest.java

```diff
@@ -0,0 +1,203 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.platform.db.migration.version.v67;
+
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Random;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import org.junit.Rule;
+import org.junit.Test;
+import org.sonar.api.utils.System2;
+import org.sonar.api.utils.internal.TestSystem2;
+import org.sonar.db.CoreDbTester;
+
+import static org.apache.commons.lang.RandomStringUtils.randomAlphanumeric;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class CleanupDisabledUsersTest {
+
+  private final static long PAST = 100_000_000_000L;
+  private final static long NOW = 500_000_000_000L;
+  private final static Random RANDOM = new Random();
+  private final static String SELECT_USERS = "select name, scm_accounts, user_local, login, crypted_password, salt, email, external_identity, external_identity_provider, active, is_root, onboarded, created_at, updated_at from users";
+
+  private System2 system2 = new TestSystem2().setNow(NOW);
+
+  @Rule
+  public CoreDbTester db = CoreDbTester.createForSchema(CleanupDisabledUsersTest.class, "users.sql");
+
+  private CleanupDisabledUsers underTest = new CleanupDisabledUsers(db.database(), system2);
+
+  private final List<User> users = Arrays.asList(
+    new User("user1", null, null, null, null, null, false),
+    new User("user2", randomAlphanumeric(10), null, null, null, null, false),
+    new User("user3", null, randomAlphanumeric(10), null, null, null, false),
+    new User("user4", null, null, randomAlphanumeric(10), null, null, false),
+    new User("user5", null, null, null, randomAlphanumeric(10), null, false),
+    new User("user6", null, null, null, null, randomAlphanumeric(10), false),
+    new User("user7", randomAlphanumeric(10), randomAlphanumeric(10), randomAlphanumeric(10), randomAlphanumeric(10), randomAlphanumeric(10), false),
+    new User("user8", randomAlphanumeric(10), randomAlphanumeric(10), randomAlphanumeric(10), randomAlphanumeric(10), randomAlphanumeric(10), true),
+    new User("user9", randomAlphanumeric(10), randomAlphanumeric(10), randomAlphanumeric(10), randomAlphanumeric(10), randomAlphanumeric(10), true)
+  );
+
+  @Test
+  public void do_nothing_when_no_data() throws SQLException {
+    underTest.execute();
+  }
+
+  @Test
+  public void execute_must_update_database() throws SQLException {
+    users.forEach(User::insert);
+
+    underTest.execute();
+
+    assertDatabaseContainsExpectedValues();
```

Comment:
Same comment as for the shared usage of list of users : it's better to clearly display what is the output when expect instead of using a shared method